### PR TITLE
Move logic out of CMakeLists.txt

### PR DESCRIPTION
**Pull request by ezzieyguywuf**
Thank you for creating a pull request to contribute to FreeCAD! To ease integration, please confirm the following:

- [x] Branch rebased on latest master `git pull --rebase upstream master`
- [x] Unit tests confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [x] Commit message is [well-written](https://chris.beams.io/posts/git-commit/)
- X (not applicable) Commit message includes `issue #<id>` or `fixes #<id>` where `<id>` is the [associated MantisBT](https://freecadweb.org/wiki/tracker#GitHub_and_MantisBT) issue id if one exists

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.19 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=34586).

---

I am still compiling the code - I'll update this post once I've run all tests successfully.

### CODE REVIEW

File: cMake/FreeCAD_Helpers/FindSalomeMESH.cmake [hunk 1]
```diff
@@ -0,0 +1,126 @@
+macro(FindSalomeMESH)
+# -------------------------------- Salome SMESH --------------------------
+
+    # Salome SMESH sources are under src/3rdParty now
+    if(BUILD_SMESH)
+        # set the internal smesh version:
+        # see src/3rdParty/salomonemesh/CMakeLists.txt and commit https://github.com/FreeCAD/FreeCAD/commit/666a3e5 and https://forum.freecadweb.org/viewtopic.php?f=10&t=30838
```
Comment (luzpaz) [inline cMake/FreeCAD_Helpers/FindSalomeMESH.cmake]: s/salomonemsh/salomesmesh/

File: cMake/FreeCAD_Helpers/FindSalomeMESH.cmake [hunk 2]
```diff
@@ -0,0 +1,126 @@
+macro(FindSalomeMESH)
```
Comment (luzpaz) [inline cMake/FreeCAD_Helpers/FindSalomeMESH.cmake]: Should be `FindSalomeSMESH` (also the file name should be renamed)

File: cMake/FreeCAD_Helpers/FindSalomeMESH.cmake [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+                              "================\n")
+            endif()
+            include_directories(${SMESH_INCLUDE_DIR})
+        endif()
+
+        set(SMESH_FOUND TRUE)
+        configure_file(SMESH_Version.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/SMESH_Version.h)
+    endif(BUILD_SMESH)
+
+endmacro(FindSalomeMESH)
```
Comment (luzpaz) [inline cMake/FreeCAD_Helpers/FindSalomeMESH.cmake]: s/FindSalomeMESH/FindSalomeSMESH/

File: cMake/FreeCAD_Helpers/README.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
+This folder will contain an individual cmake file for each FreeCAD "helper".
+
+A "helper" should be a macro or functio that tries (as much as possible) to adhere to the
```
Comment (luzpaz) [inline cMake/FreeCAD_Helpers/README.md]: s/functio/function/

File: cMake/FreeCAD_Helpers/README.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+UNIX philosophy - in other words, it should strive to do one thing and do it well.
+
+The idea here is to break up the cmake build system into smaller, more manageable chunks.
+This should make maintenance easier, and should also make troubleshooting a bit less
+painful. Finally, it should also clean up our top-level CMakeLists.txt file a bit, making
+it bit easier for new developpers to jump in and see what's what.
```
Comment (luzpaz) [inline cMake/FreeCAD_Helpers/README.md]: s/developpers/developers/

File: cMake/FreeCAD_Helpers/FindFreeType.cmake [hunk 6]
```diff
@@ -0,0 +1,13 @@
+macro(FindFreetype)
+#--------------------FreeType-----------------------
+
+    if(FREECAD_USE_FREETYPE)
+        find_package(Freetype)
+        if(NOT FREETYPE_FOUND)
+            message("===============================================================\n"
+                    "FreeType2 not found. Part module will lack of makeWireString().\n"
```
Comment (luzpaz) [inline cMake/FreeCAD_Helpers/FindFreeType.cmake]: `"FreeType2 not found. Part module will lack functionality of makeWireString().\n"`

File: cMake/FreeCAD_Helpers/FindSalomeMESH.cmake (hunk 1 above)
Comment (ezzieyguywuf) [reply cMake/FreeCAD_Helpers/FindSalomeMESH.cmake]: Good catch, I made this change locally but failed to add it to the git repo.

Comment (ezzieyguywuf) [reply cMake/FreeCAD_Helpers/FindSalomeMESH.cmake]: The "salomonemsh" error is [in master](https://github.com/FreeCAD/FreeCAD/blob/master/CMakeLists.txt#L680). For this initial PR, I wanted to literally copy/paste what is currently in master and simply move it to the cMake sub-directory. In a subsequent future PR I will start going through the actual code/logic and try to clean things like this up.

File: cMake/FreeCAD_Helpers/FindSalomeMESH.cmake (hunk 2 above)
Comment (ezzieyguywuf) [reply cMake/FreeCAD_Helpers/FindSalomeMESH.cmake]: Good catch, I made this change locally but failed to add it to the git repo.

File: cMake/FreeCAD_Helpers/FindFreeType.cmake (hunk 6 above)
Comment (ezzieyguywuf) [reply cMake/FreeCAD_Helpers/FindFreeType.cmake]: Again, this error is currently [in master ](https://github.com/FreeCAD/FreeCAD/blob/master/CMakeLists.txt#L963). I would like to leave it as-is for this PR.

**Comment by ezzieyguywuf** [comment]
I can confirm that all unit tests now pass on my local system.

**Comment by wwmayer** [comment]
https://forum.freecadweb.org/viewtopic.php?f=17&t=39067

**Comment by wwmayer** [comment]
If you look on the travis builds then you will realize that it doesn't work on Windows. It cannot find the Qt5 modules.

**Comment by ezzieyguywuf** [comment]
@wwmayer I'm aware of the travis issues on Windows. I am working on setting up a build environment in Windows in order to try to duplicate. In the meantime, I've merged in the latest master.

**Comment by ezzieyguywuf** [comment]
@wwmayer @luzpaz it seems that this PR is now passing the travis builds. I have squashed the commits back down to one and rebased over the latest master.

**Comment by wwmayer** [comment]
Merged.

**Comment by luzpaz** [comment]
Quick request, could we add a comment in the top level CMakeList.txt about we've moved most CMake 'modules' in to `./cMake/FreeCAD_Helpers/` ? or is it superfluous ?

**Comment by ezzieyguywuf** [comment]
I can add a comment. I was going to prepare some new PRs with cmake updates
/ fixes I’ll put it in one of those

**Pull request closed without merging**